pilot/pkg/bootstrap/server.go

```diff
@@ -575,29 +549,68 @@ func (s *Server) createK8sServiceControllers(serviceControllers *aggregate.Contr
 	return
 }
 
-// initMultiClusterController initializes multi cluster controller
-// currently implemented only for kubernetes registries
-func (s *Server) initMultiClusterController(args *PilotArgs) error {
+// initMultiClusterSecretController initializes multi cluster secret controller
+// to watch for remote cluster additions or deletions.  The secret controller
+// will call back to functions defined here to add handlers to watch remote clusters
+// Currently implemented only for kubernetes registries.
+func (s *Server) initMultiClusterSecretController(args *PilotArgs) (err error) {
 	if hasKubeRegistry(args) {
-		s.addStartFunc(func(stop <-chan struct{}) error {
-			secretController := clusterregistry.NewController(s.kubeClient,
-				args.Config.ClusterRegistriesNamespace,
-				s.clusterStore,
-				s.ServiceController,
-				s.EnvoyXdsServer,
-				args.Config.ControllerOptions.ResyncPeriod,
-				args.Config.ControllerOptions.WatchedNamespace,
-				args.Config.ControllerOptions.DomainSuffix)
-
-			// Start secret controller which watches for runtime secret Object changes and adds secrets dynamically
-			go secretController.Run(stop)
 
-			return nil
-		})
+		err = secretcontroller.StartSecretController(s.kubeClient, s.addMemberCluster, s.deleteMemberCluster, args.Config.ClusterRegistriesNamespace)
+		if err != nil {
+			return fmt.Errorf("could not start secret controller: %v", err)
+		}
+
 	}
 	return nil
 }
 
+//addMemberCluster is passed to the secret controller as a callback to be called
+//when a remote cluster is added.  This function needs to set up all the handlers
+//to watch for resources being added, deleted or changed on remote clusters.
```

Comment:
white space

Reply:
Ask for white space changes that the linter doesn't flag seems a bit petty - but I fixed them up